### DISCUSSION TITLE
yolov10x-seg

### DISCUSSION THREAD

**Discussion by AISTALK** [ultralytics/ultralytics#15585] [Q&A]
I have seen yolov9l-seg and yolov9x-seg in ultralytics assets.

Am I understanding right that to do the yolov10x-seg model we should simply could take the segmentation head of yolov9 and backbone of yolov10?

**Comment by glenn-jocher** [comment]
@AISTALK thank you for your interest in YOLOv10. The YOLOv10 architecture introduces several innovations, including NMS-free training and a holistic efficiency-accuracy driven design. While combining the segmentation head of YOLOv9 with the backbone of YOLOv10 might seem feasible, it is not officially supported and may not leverage the full benefits of YOLOv10's optimizations. For the best performance, we recommend using the models as provided. If you have further questions, feel free to ask!

**Reply by AISTALK** [reply]
Does the ultralytics team work on creating yolov10*-seg series?

**Reply by glenn-jocher** [reply]
@AISTALK thank you for your interest! Currently, there are no official plans to release a YOLOv10-seg series. We recommend using the existing YOLOv9-seg models for segmentation tasks. If you have any other questions, feel free to ask!

**Reply by AISTALK** [reply]
> @AISTALK thank you for your interest! Currently, there are no official plans to release a YOLOv10-seg series. We recommend using the existing YOLOv9-seg models for segmentation tasks. If you have any other questions, feel free to ask!

How could I contribute and make yolov10*-seg?

**Reply by glenn-jocher** [reply]
Thank you for your enthusiasm! You can contribute by forking the repository, implementing the YOLOv10-seg model, and submitting a pull request. Please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing) for detailed instructions.

**Reply by JHLZXB** [reply]
I have a questions: can yolov8-seg or yolov11-seg series  be used in yolov10 to train？
